packages/synapse-react-client/src/synapse-client/SynapseClient.ts

```diff
@@ -2277,14 +2289,15 @@ const uploadFilePart = async (
   file: Blob,
   contentType: string,
   getIsCancelled?: () => boolean,
+  abortController?: AbortController,
 ) => {
   const controller = new AbortController()
-  const signal = controller.signal
+  const signal = abortController?.signal || controller.signal
 
   const checkIsCancelled = () => {
     if (getIsCancelled) {
       const isCancelled = getIsCancelled()
-      if (isCancelled) {
+      if (isCancelled && !controller.signal.aborted) {
```

Comment:
should this be `!signal.aborted` rather than `!controller.signal.aborted`?

Reply:
Yes, nice catch. Thanks!